File: problems/2163-kth-distinct-string-in-an-array/kth-distinct-string-in-an-array.py
class Solution:
    def kthDistinct(self, arr: List[str], k: int) -> str:
        count = Counter(arr)
        distinct = []
        for cha in arr:
            if count[cha] == 1:
                distinct.append(cha)
        result = ""
        if len(distinct) >= k:
            result = distinct[k - 1]
        
        return result

File: problems/2163-kth-distinct-string-in-an-array/kth-distinct-string-in-an-array.js
/**
 * @param {string[]} arr
 * @param {number} k
 * @return {string}
 */
const kthDistinct = (arr, k) => {
   const countMap = new Map();
   for (let cha of arr) {
     countMap.set(cha, countMap.get(cha) ? countMap.get(cha) + 1 : 1) 
   }

   let distinct = []
   for (let [key, value] of countMap.entries()){
        if (value === 1) {
            distinct = [...distinct, key]
        }
   }
   return distinct[k - 1] ? distinct[k - 1] : ""
};